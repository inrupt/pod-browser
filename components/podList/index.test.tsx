/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";

import PodList from "./index";

describe("Pod list", () => {
  test("Renders null if there are no pod iris", () => {
    const iris = undefined;
    const tree = shallow(<PodList podIris={iris} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  test("Renders a container if there is one pod iri", () => {
    const iris = ["https://mypod.myhost.com"];
    const tree = shallow(<PodList podIris={iris} />);

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  test("Renders a table if there are multiple pod iris", () => {
    const iris = [
      "https://mypod.myhost.com",
      "https://myotherpod.myotherhost.com",
    ];

    const tree = shallow(<PodList podIris={iris} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
