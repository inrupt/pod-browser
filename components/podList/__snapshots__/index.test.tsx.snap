// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pod list Renders a container if there is one pod iri 1`] = `
<Container
  iri="https://mypod.myhost.com"
/>
`;

exports[`Pod list Renders a table if there are multiple pod iris 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Table))>
    <WithStyles(ForwardRef(TableHead))>
      <WithStyles(ForwardRef(TableRow))>
        <WithStyles(ForwardRef(TableCell))>
          Pod IRI
        </WithStyles(ForwardRef(TableCell))>
      </WithStyles(ForwardRef(TableRow))>
    </WithStyles(ForwardRef(TableHead))>
    <WithStyles(ForwardRef(TableBody))>
      <ContainerTableRow
        iri="https://mypod.myhost.com"
        key="https://mypod.myhost.com"
      />
      <ContainerTableRow
        iri="https://myotherpod.myotherhost.com"
        key="https://myotherpod.myotherhost.com"
      />
    </WithStyles(ForwardRef(TableBody))>
  </WithStyles(ForwardRef(Table))>
  <DetailsContextMenu />
</Fragment>
`;

exports[`Pod list Renders null if there are no pod iris 1`] = `""`;
